Doppler.CDHelper.Test/DockerHubHandlerTest.cs

```diff
@@ -0,0 +1,106 @@
+
+        [Fact]
+        public async Task POST_dockerhub_handler_should_parse_and_log_callback_url()
+        {
+            // Arrange
+            var secret = "my_secret";
+            var callbackUrl = "https://registry.hub.docker.com/u/dopplerdock/doppler-cd-helper/hook/2jiedged52hbhfi1acgdggdi1ih123456/";
+            var a = (Microsoft.Extensions.Logging.Logger<Doppler.CDHelper.Controllers.HooksController>)null;
+
+            var loggerMock = new Mock<ILogger<HooksController>>();
+            // loggerMock.Setup(x => x.LogInformation(It.IsAny<EventId>(), It.IsAny<Exception>(), It.IsAny<string>(), It.IsAny<object[]>()))
+            //     .;
+
+            using var f = _factory.WithWebHostBuilder(c =>
+            {
+                c.ConfigureServices(s =>
+                {
+                    s.AddSingleton(loggerMock.Object);
+                });
+            });
+
+
+            var client = f.CreateClient(new WebApplicationFactoryClientOptions()
+            {
+                AllowAutoRedirect = false
+            });
+
+            // Act
+            var response = await client.PostAsync(
+                $"/hooks/{secret}/",
+                JsonContent.Create(new
+                {
+                    push_data = new
+                    {
+                        pushed_at = 1618614391,
+                        images = new object[0],
+                        tag = "pr-1",
+                        pusher = "dopplerdock"
+                    },
+                    callback_url = callbackUrl,
+                    repository = new
+                    {
+                        status = "Active",
+                        description = "",
+                        is_trusted = false,
+                        full_description = (string)null,
+                        repo_url = "https://hub.docker.com/r/dopplerdock/doppler-cd-helper",
+                        owner = "dopplerdock",
+                        is_official = false,
+                        is_private = true,
+                        name = "doppler-cd-helper",
+                        @namespace = "dopplerdock",
+                        star_count = 0,
+                        comment_count = 0,
+                        date_created = 1618603558,
+                        repo_name = "dopplerdock/doppler-cd-helper"
+                    }
+                }
+            ));
+
+            // Assert
+
+            // It is a FormattedLogValues object
+            // See: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/FormattedLogValues.cs#L16
+            IReadOnlyList<KeyValuePair<string, object>> logParameters = null;
+
+            loggerMock.Verify(
+                x => x.Log(
+                    It.IsAny<LogLevel>(),
+                    It.IsAny<EventId>(),
+                    It.Is<It.IsAnyType>((v, t) => AssertHelper.IsTypeAndGetValue(v, out logParameters)),
```

Comment:
It is not so elegant, but I cannot find a better way to test it.

```
                    It.Is<It.IsAnyType>((v, t) => AssertHelper.IsTypeAndGetValue(v, out logParameters)),
                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Do you have any idea?